src/editors/containers/VideoUploadEditor/index.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDropzone } from 'react-dropzone';
+import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Icon } from '@edx/paragon';
+import './index.scss';
+import { useDispatch } from 'react-redux';
+import { ArrowForward, FileUpload } from '@edx/paragon/icons';
+import * as hooks from './hooks';
+import messages from '../../messages';
+
+export const VideoUploader = ({ onUpload, errorMessage }) => {
+  const [, setUploadedFile] = useState();
+  const [textInputValue, setTextInputValue] = useState('');
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    accept: 'video/*',
+    multiple: false,
+    onDrop: (acceptedFiles) => {
+      if (acceptedFiles.length > 0) {
+        const uploadfile = acceptedFiles[0];
+        setUploadedFile(uploadfile);
+        onUpload(uploadfile);
```

Comment:
@pkulkark Can this create a scenario when the file is set, but hasn't been uploaded? I guess it's ok if the parent component displays the error, and the user can remove the selected file and choose a different one. Just pointing it out, in case it's not an intended behavior.

Reply:
This will be needed when we implement the displaying part. I think it requires a thumbnail of the video to be shown which should be handled in a later task.